### PR TITLE
[expo] Upgrade `react-native-reanimated` to `3.10.0`

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Upgrades `react-native-reanimated` to `3.10.0`

# Test Plan

- Expo Go ✅ 
- Bare expo ✅

### CODE REVIEW

File: apps/expo-go/ios/Podfile.lock [hunk 1]
```diff
@@ -2399,7 +2399,7 @@ SPEC CHECKSUMS:
   GoogleDataTransport: 54dee9d48d14580407f8f5fbf2f496e92437a2f2
   GoogleMaps: 032f676450ba0779bd8ce16840690915f84e57ac
   GoogleUtilities: 13e2c67ede716b8741c7989e26893d151b2b2084
-  hermes-engine: 118ab4d39602f2b05fd37ba7356343d2b5344b9c
+  hermes-engine: 1d242e1d7d6c63ee223040c5a0d72d59105ed811
```
Comment (tsapeta) [inline apps/expo-go/ios/Podfile.lock:2402]: I think this hash is not correct. It seems to be unstable again... 😢 Probably better to just revert this change.

**Review by tsapeta — APPROVED** [review]

File: apps/bare-expo/ios/Podfile.lock [hunk 2]
```diff
@@ -2641,7 +2641,7 @@ SPEC CHECKSUMS:
   RNDateTimePicker: cd42eda5f315fc320f0b359413bd598957f7e601
   RNFlashList: b521ebdd7f9352673817f1d98e8bdc0c8cf8545b
   RNGestureHandler: 4d271a5d9178403bd8dae0d853e2be21cf53743c
-  RNReanimated: b0e16e5938d510b2203a2195c4f1563de2a23e49
+  RNReanimated: c90868a27d5b89bbd35a7e3b048fefe4c9706239
```
Comment (brentvatne) [inline apps/bare-expo/ios/Podfile.lock:2644]:
```suggestion
  RNReanimated: b0e16e5938d510b2203a2195c4f1563de2a23e49
```
